ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -200,6 +211,7 @@ public void onClose(final Session session, final CloseReason closeReason) {
     if (subscriber != null) {
       subscriber.close();
     }
+    securityExtension.close();
```

Comment:
Not quite understanding your concern. Where in this method are we doing anything with the `serviceContext`? Or are you saying the problem is that we aren't doing anything with the `serviceContext` in this method though we should be?

Reply:
Aaah, well, first I just got confused with the name :P. Sorry hehe. Second, yes, we should close the `ServiceContext` created when the session was opened, otherwise, we'll leak `AdminClient` connections I think.

Reply:
I'll have to investigate more to understand whether we're currently leaking the `ServiceContext` or not. It seems outside the scope of this PR so I'm inclined to do it in a follow-up instead.

Reply:
ok, we can figure out about leaking a service context later. However, we should remove this `securityExtension.close()` line. This `close` is called when the KSQL server is stopped (see KsqlRestApplication#stop()). It needs to be called only once instead of per session.

Reply:
Good catch! Fixed.